spec/facility_spec.rb

```diff
@@ -2,25 +2,201 @@
     end
   end
 
   describe '#add service' do
     it 'can add available services' do
-      expect(@facility.services).to eq([])
-      @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
-      @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      expect(@facility_1.services).to eq([])
+      @facility_1.add_service('New Drivers License')
+      expect(@facility_1.services).to eq(['New Drivers License'])
+      @facility_1.add_service('Renew Drivers License')
+      expect(@facility_1.services).to eq(['New Drivers License', 'Renew Drivers License'])
+      @facility_1.add_service('Vehicle Registration')
+      expect(@facility_1.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+    end
+  end
+
+  describe '#register_vehicle' do
+    it 'can register vehicles' do
+      @facility_1.add_service('Vehicle Registration')
+      expect(@cruz.registration_date).to be nil
+      expect(@facility_1.registered_vehicles).to eq([])
+      expect(@facility_1.collected_fees).to eq(0)
+
+      @facility_1.register_vehicle(@cruz)
+      expect(@cruz.registration_date).to eq(Date.today)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+      expect(@facility_1.collected_fees).to eq(100)
+
+      @facility_1.register_vehicle(@camaro)
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro])
+      expect(@facility_1.collected_fees).to eq(125)
+
+      @facility_1.register_vehicle(@bolt)
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+
+    it 'does nothing if there is no Vehicle Registration service available' do
+      expect(@facility_1.services.include?('Vehicle Registration')).to be false
+      expect(@facility_1.registered_vehicles).to eq([])
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.registered_vehicles).to eq([])
+    end
+  end
+
+  describe '#collect_registration_fee' do
+    it 'can collect registration fee' do
+      @facility_1.add_service('Vehicle Registration')
+      expect(@facility_1.collected_fees).to eq(0)
+      @facility_1.collect_registration_fee(@cruz)
+      expect(@facility_1.collected_fees).to eq(100)
+      @facility_1.collect_registration_fee(@bolt)
+      expect(@facility_1.collected_fees).to eq(300)
+      @facility_1.collect_registration_fee(@camaro)
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+  end
+
+  describe '#administer_written_test' do
+    it 'can administer written test if age >= 16 and has a permit' do
```

Comment:
Your test names are wonderful! Short, succinct, yet very accurate of what the test is looking for